kotlinx-coroutines-core/common/src/internal/ConcurrentLinkedList.kt

```diff
@@ -207,21 +262,20 @@ internal abstract class Segment<S : Segment<S>>(
     abstract val numberOfSlots: Int
 
     /**
-     * Numbers of cleaned slots (the lowest bits) and AtomicRef pointers to this segment (the highest bits)
+     * Number of cleaned slots.
      */
-    private val cleanedAndPointers = atomic(pointers shl POINTERS_SHIFT)
+    private val cleanedSlots = atomic(0)
 
     /**
      * The segment is considered as removed if all the slots are cleaned
      * and there are no pointers to this segment from outside.
      */
-    override val isRemoved get() = cleanedAndPointers.value == numberOfSlots && !isTail
+    override val isRemoved get() = cleanedSlots.value == numberOfSlots && !isTail
 
-    // increments the number of pointers if this segment is not logically removed.
-    internal fun tryIncPointers() = cleanedAndPointers.addConditionally(1 shl POINTERS_SHIFT) { it != numberOfSlots || isTail }
-
-    // returns `true` if this segment is logically removed after the decrement.
-    internal fun decPointers() = cleanedAndPointers.addAndGet(-(1 shl POINTERS_SHIFT)) == numberOfSlots && !isTail
+    /**
+     * Shows if all nodes going before this node have been processed.
```

Comment:
Please improve the clarity of the documentation here and provide an illustrative example.